### PR TITLE
Documentation for 2FA

### PR DESCRIPTION
**Pull request by ketankartoza**
This PR is for issue #1457

### CODE REVIEW

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 2 min 6 sec

File: docs/src/user/manual/register/setup-2fa.md [hunk 1]
```diff
@@ -1,2 +1,94 @@
...
+    * The authenticator app will now generate one-time codes for SAWPS account.
+
+3. **Step 2**: Input the Token
+
+    * After scanning the QR code, user will see user's SAWPS account added to your authenticator app.
+    * The app will continuously generate six-digit codes that change every 30 seconds.
+    * To enable 2FA, user need to input a generated token from the app.
+    * Input the current six-digit token generated by authenticator app into the input field named `Token`.
+
+4. **VERIFY**: User can verify token using the `VERIFY` button. Upon successful verification, Two-Factor Authentication will be enabled for SAWPS account.
```
Comment (Jeremy-Prior) [inline docs/src/user/manual/register/setup-2fa.md:32]:
```suggestion
4. **VERIFY**: Users can verify the token using the `VERIFY` button. Upon successful verification, Two-Factor Authentication will be enabled for SAWPS account and the user will be redirected to continue registering.
```

File: docs/src/user/manual/register/setup-2fa.md [hunk 2]
```diff
@@ -1,2 +1,94 @@
...
+
+# Two-Factor Authentication (2FA) Once Verify
+
+Once user will successfully verify the 2FA, user will able to add device, delete device and generate backop tokens.
+
+![Two-Factor Authentication (2FA) Once Verify](./img/setup-2fa-2.png)
+
+1. **Two-factor authentication method**: Displays the mathod used for the 2FA.
+
+2. **Two-factor authentication devices**: Two-factor authentication devices table displays names of devices and delete icon to delete the specific device.
```
Comment (Jeremy-Prior) [inline docs/src/user/manual/register/setup-2fa.md:42]:
```suggestion
2. **Two-factor authentication devices**: The two-factor authentication devices table displays the names of devices and the delete icon to delete the specific device.
```

File: docs/src/user/manual/register/setup-2fa.md [hunk 3]
```diff
@@ -1,2 +1,94 @@
...
+
+    ![Two-factor authentication devices table](./img/setup-2fa-5.png)
+
+    1. **Device Name**: Displays name of added devices.
+
+    2. **Delete Icon**: User can delete the specific device by clicking on the delete icon.
```
Comment (Jeremy-Prior) [inline docs/src/user/manual/register/setup-2fa.md:48]:
```suggestion
    2. **Delete Icon**: Users can delete the specific device by clicking on the delete icon.
```

File: docs/src/user/manual/register/setup-2fa.md [hunk 4]
```diff
@@ -1,2 +1,94 @@
...
+
+        By clicking on the delete icon user will see popup for device delete confirmation.
```
Comment (Jeremy-Prior) [inline docs/src/user/manual/register/setup-2fa.md:50]:
```suggestion
        By clicking on the delete icon users will see popup for confirming the deletion of a device.
```

File: docs/src/user/manual/register/setup-2fa.md [hunk 5]
```diff
@@ -1,2 +1,94 @@
...
+
+        ![Delete Confirmation](./img/setup-2fa-6.png)
+
+        1. **Cancel**: `CANCEL` button to cancel device deletion.
+
+        2. **Delete**: `DELETE` button to delete device.
```
Comment (Jeremy-Prior) [inline docs/src/user/manual/register/setup-2fa.md:56]:
```suggestion
        1. The `CANCEL` button is used to cancel the device deletion.

        2. The `DELETE` button is used to delete the device.
```

File: docs/src/user/manual/register/setup-2fa.md [hunk 6]
```diff
@@ -1,2 +1,94 @@
...
+
+3. **Add Device**: `ADD DEVICE` button to add new device. By clicking on the `ADD DEVICE` button user will see the popup to add new device.
```
Comment (Jeremy-Prior) [inline docs/src/user/manual/register/setup-2fa.md:58]:
```suggestion
3. Click on the `ADD DEVICE` button to add a new device. By clicking on the button users will see the popup to add new device.
```

File: docs/src/user/manual/register/setup-2fa.md [hunk 7]
```diff
@@ -1,2 +1,94 @@
...
+
+    ![Add Device](./img/setup-2fa-3.png)
+
+    1. **Device Name**: Input field to insert device name.
```
Comment (Jeremy-Prior) [inline docs/src/user/manual/register/setup-2fa.md:62]:
```suggestion
    1. **Device Name**: Input field to add device name.
```

File: docs/src/user/manual/register/setup-2fa.md [hunk 8]
```diff
@@ -1,2 +1,94 @@
...
+
+        ![QR Scan](./img/setup-2fa-4.png)
+
+        By clicking on the `ADD` button user will see popup for scaning the QR code. Scan QR code to generate token for new added device.
+
+        1. **QR Code**: QR code to scan and generate token.
+
+        2. **Close**: `CLOSE` button to close the popup.
+
+    3. **Cancel**: `CANCEL` button to cancel adding new device process.
```
Comment (Jeremy-Prior) [inline docs/src/user/manual/register/setup-2fa.md:74]:
```suggestion
    3. **Cancel**: `CANCEL` button to cancel the process of adding a new device.
```

File: docs/src/user/manual/register/setup-2fa.md [hunk 9]
```diff
@@ -1,2 +1,94 @@
...
+
+    2. **ADD**: `ADD` button to add new device.
+
+        ![QR Scan](./img/setup-2fa-4.png)
+
+        By clicking on the `ADD` button user will see popup for scaning the QR code. Scan QR code to generate token for new added device.
+
+        1. **QR Code**: QR code to scan and generate token.
+
+        2. **Close**: `CLOSE` button to close the popup.
```
Comment (Jeremy-Prior) [inline docs/src/user/manual/register/setup-2fa.md:72]:
```suggestion
    2. *By clicking on the `ADD` button users will be shown a popup for scanning the QR code. Users can scan QR code to generate token for new added device.

        ![QR Scan](./img/setup-2fa-4.png)

        1. **QR Code**: to be scanned to generate a token.

        2. `CLOSE` button to close the popup.
```

File: docs/src/user/manual/register/setup-2fa.md [hunk 10]
```diff
@@ -1,2 +1,94 @@
...
+
+    3. **Cancel**: `CANCEL` button to cancel adding new device process.
+
+4. **Two-factor authentication recovery tokens**: Information text for token generation.
+
+5. **Generate Backup Tokens**: By clicking on the `GENERATE BACKUP TOKENS` button user can generate tokens.
```
Comment (Jeremy-Prior) [inline docs/src/user/manual/register/setup-2fa.md:78]:
```suggestion
5. **`GENERATE BACKUP TOKENS`**: By clicking on this button users can generate their recovery tokens.
```

File: docs/src/user/manual/register/setup-2fa.md [hunk 11]
```diff
@@ -1,2 +1,94 @@
...
+
+6. **Profile**: The `PROFILE` navigation link navigates to the [profile page](../user-profile/profile-page.md).
```
Comment (Jeremy-Prior) [inline docs/src/user/manual/register/setup-2fa.md:80]:
```suggestion
6. The `PROFILE` navigation link navigates to the [profile page](../user-profile/profile-page.md).
```

File: docs/src/user/manual/register/setup-2fa.md [hunk 12]
```diff
@@ -1,2 +1,94 @@
...
+
+7. **Reminders**: The `REMINDERS` navigation link navigates to the [reminders page](../user-profile/reminders.md).
```
Comment (Jeremy-Prior) [inline docs/src/user/manual/register/setup-2fa.md:82]:
```suggestion
7. The `REMINDERS` navigation link navigates to the [reminders page](../user-profile/reminders.md).
```

File: docs/src/user/manual/register/setup-2fa.md [hunk 13]
```diff
@@ -1,2 +1,94 @@
...
+
+8. **Notifications**:  The `NOTIFICATIONS` navigation link navigates to the [notification page](../user-profile/notifications.md).
```
Comment (Jeremy-Prior) [inline docs/src/user/manual/register/setup-2fa.md:84]:
```suggestion
8. The `NOTIFICATIONS` navigation link navigates to the [notification page](../user-profile/notifications.md).
```

File: docs/src/user/manual/register/setup-2fa.md [hunk 14]
```diff
@@ -1,2 +1,94 @@
...
+
+9. **Organisations**: The `ORGANISATIONS` navigation link navigates to the [organisation page](../user-profile/organisation-page.md) for registered users.
```
Comment (Jeremy-Prior) [inline docs/src/user/manual/register/setup-2fa.md:86]:
```suggestion
9. The `ORGANISATIONS` navigation link navigates to the [organisation page](../user-profile/organisation-page.md) for registered users.
```

File: docs/src/user/manual/register/setup-2fa.md [hunk 15]
```diff
@@ -1,2 +1,94 @@
...
+
+## Using Two-Factor Authentication
+
+From now on, when user log in to SAWPS account, user will be prompted to enter the six-digit token generated by user's authenticator app in addition to user's regular login credentials. This extra layer of security will help protect user's account from unauthorized access.
```
Comment (Jeremy-Prior) [inline docs/src/user/manual/register/setup-2fa.md:90]:
```suggestion
When users log in to SAWPS account, they will be prompted to enter the six-digit token generated by the user's authenticator application in addition to their regular login credentials. This extra layer of security will help protect the user's account from unauthorized access.
```

File: docs/src/user/manual/register/setup-2fa.md [hunk 16]
```diff
@@ -1,2 +1,94 @@
...
+
+## Conclusion
+
+Congratulations! You have successfully set up Two-Factor Authentication (2FA) for your SAWPS account, enhancing the security of your data and account. If you encounter any issues during the setup process, please reach out to our support team for assistance.
```
Comment (Jeremy-Prior) [inline docs/src/user/manual/register/setup-2fa.md:94]:
```suggestion
If users encounter any issues during the setup process, they should reach out to the support team for assistance.
```

**Review by Jeremy-Prior — CHANGES_REQUESTED** [review]

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 1 min 40 sec

File: docs/src/user/manual/register/setup-2fa.md [hunk 17]
```diff
@@ -1,2 +1,77 @@
-# Set up Two Factor Authentication
-<!-- To Be Populated -->
+# Two-Factor Authentication (2FA) Setup Guide
```
Comment (Jeremy-Prior) [inline docs/src/user/manual/register/setup-2fa.md:1]:
```suggestion
# Two-Factor Authentication (2FA) Setup Page
```

File: docs/src/user/manual/register/setup-2fa.md [hunk 18]
```diff
@@ -1,2 +1,77 @@
...
+
+![TWO-FACTOR-AUTHENTICATION](./img/setup-2fa-1.png)
+
+1. **Two-Factor Authentication**: Users can generate and verify the token on this page.
+
+2. **Step 1**: Users will see a QR code displayed on the screen that they can scan using their preferred authenticator application.
+3. **Step 2**: The `Token` field is where users should type in the 6-digit token (generated by their authenticator application) that is associated with the account they are registering.
+4. **VERIFY**: Users can verify the token using the `VERIFY` button. Upon successful verification, Two-Factor Authentication will be enabled for SAWPS account and the user will be redirected to continue registering.
+
+# Two-Factor Authentication (2FA) Once Verify
```
Comment (Jeremy-Prior) [inline docs/src/user/manual/register/setup-2fa.md:21]:
```suggestion
# Two-Factor Authentication (2FA) Once Verified
```

File: docs/src/user/manual/register/setup-2fa.md [hunk 19]
```diff
@@ -1,2 +1,77 @@
...
+
+Once users have successfully verified the 2FA, they will able to add device, delete device and generate backup tokens.
```
Comment (Jeremy-Prior) [inline docs/src/user/manual/register/setup-2fa.md:23]:
```suggestion
Once users have successfully verified the 2FA, they will be able to add device, delete device and generate backup tokens.
```

File: docs/src/user/manual/register/setup-2fa.md [hunk 20]
```diff
@@ -1,2 +1,77 @@
...
+
+        2. The `DELETE` button is used to delete the device.
+
+3. Click on the `ADD DEVICE` button to add a new device. By clicking on the button users will see the popup to add new device.
+
+    ![Add Device](./img/setup-2fa-3.png)
+
+    1. **Device Name**: Input field to add device name.
+
+    2. *By clicking on the `ADD` button users will be shown a popup for scanning the QR code. Users can scan QR code to generate token for new added device.
```
Comment (Jeremy-Prior) [inline docs/src/user/manual/register/setup-2fa.md:51]:
```suggestion
    2. *By clicking on the `ADD` button users will be shown a popup for scanning the QR code. Users can scan the QR code to generate a token for a newly added device.
```

File: docs/src/user/manual/register/setup-2fa.md [hunk 21]
```diff
@@ -1,2 +1,77 @@
...
+
+7. The `REMINDERS` navigation link navigates to the [reminders page](../user-profile/reminders.md).
+
+8. The `NOTIFICATIONS` navigation link navigates to the [notification page](../user-profile/notifications.md).
+
+9. The `ORGANISATIONS` navigation link navigates to the [organisation page](../user-profile/organisation-page.md) for registered users.
+
+## Using Two-Factor Authentication
+
+When users log in to SAWPS account, they will be prompted to enter the six-digit token generated by the user's authenticator application in addition to their regular login credentials. This extra layer of security will help protect the user's account from unauthorized access.
```
Comment (Jeremy-Prior) [inline docs/src/user/manual/register/setup-2fa.md:75]:
```suggestion
When users log in to their SAWPS account, they will be prompted to enter the six-digit token generated by the user's authenticator application in addition to their regular login credentials. This extra layer of security will help protect the user's account from unauthorized access.
```

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 1 min 54 sec

**Review by Jeremy-Prior — APPROVED** [review]
LGTM

**Comment by Jeremy-Prior** [comment]
@dimasciput please merge

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 1 min 55 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 1 min 31 sec

**Pull request merged**